File: leetcode/medium/1833. Maximum Ice Cream Bars/solution.js
// https://leetcode.com/problems/maximum-ice-cream-bars/

/**
 * @param {number[]} costs
 * @param {number} coins
 * @return {number}
 */
function maxIceCream(costs, coins) {
    let i = 0;
    costs.sort((a, b) => a - b);
    while (coins > 0) {
        if (costs[i] <= coins) coins -= costs[i];
        else break;
        i += 1;
    }
    return i;
};

console.log(maxIceCream([1, 6, 3, 1, 2, 5], 20))

File: leetcode/medium/1833. Maximum Ice Cream Bars/solution.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int i = 0;
        for (i = 0; i < costs.size(); i += 1) {
            if (coins > 0 && costs[i] <= coins) coins -= costs[i];
            else break;
        }
        return i;
    }
};

int main() {
    Solution solution;
    vector<int> costs = {1, 6, 3, 1, 2, 5};
    cout << solution.maxIceCream(costs, 20);

    return 0;
}